generate-spec

```diff
@@ -712,15 +751,100 @@ if (count($schemas) == 0 && count($routes) == 0) {
 	Logger::error("app", "No spec generated");
 }
 
-if (count($openapi["paths"]) == 0) {
-	$openapi["paths"] = new stdClass();
-}
-
 ksort($schemas);
-$openapi["components"]["schemas"] = count($schemas) == 0 ? new stdClass() : $schemas;
 
 if ($useTags) {
 	$openapi["tags"] = $tags;
 }
 
-file_put_contents($out, json_encode($openapi, Helpers::jsonFlags()));
+foreach ($scopePaths as $scope => $paths) {
+	$openapiScope = $openapi;
+
+	if (count($paths) == 0) {
+		$paths = new stdClass();
+	}
+
+	$scopeSuffix = $scope === 'default' ? '' : '-' . $scope;
+	$openapiScope['info']['title'] .= $scopeSuffix;
+	$openapiScope['paths'] = $paths;
+
+	$usedSchemas = [];
+	foreach ($paths as $url => $urlRoutes) {
+		foreach ($urlRoutes as $httpMethod => $routeData) {
+			foreach ($routeData['responses'] as $statusCode => $responseData) {
+				if (isset($responseData['content']['application/json'])) {
+					if (is_array($responseData['content']['application/json']['schema'])) {
+						$newSchemas = Helpers::collectUsedRefs($responseData['content']['application/json']['schema']);
+						$usedSchemas = array_merge($usedSchemas, $newSchemas);
+					}
+				} elseif (isset($responseData['content']['*/*']['schema']['type'], $responseData['content']['*/*']['schema']['format'])
+					&& $responseData['content']['*/*']['schema']['type'] === 'string'
+					&& $responseData['content']['*/*']['schema']['format'] === 'binary') {
+					Logger::info("app", "Binary response from '$httpMethod $url' - Skipping schema reading");
+				} else {
+					Logger::info("app", "Could not read used schemas for response to '$httpMethod $url' with status code $statusCode");
+				}
+			}
+		}
+	}
```

Comment:
I feel like this is too specific and can break easily. For example it doesn't account for requestBody (which isn't used yet, but probably will be in the future). I would appreciate a more general approach that just checks everything and not the specific parts of the responses.

Reply:
Not sure how to improve this. If something else than JSON/data is returned, there can not be a ref with a schema type in it.

Can you give a sample of what you think a problem is, so we can generate a test case and look at a solution

Reply:
I would just loop all the content types too and then call collectUsedRefs on the schema. If the type is binary it will find no refs, so there shouldn't be a problem. This way the different content types are handled the same.
There are also more content types than the ones that are used here, in fact any content type can appear since some Responses allow setting it manually.

Reply:
I still don't get this. The only place this happens atm is with OCS exceptions, and that is a bug in the tool (handled in #43 ).

I improved the detection and handling in the last commit and now think this is good to go.